src/Command/ResetProductIndexCommand.php

```diff
@@ -62,43 +63,54 @@ protected function configure(): void
+        if ($lockHandler->lock()) {
 
-            return;
-        }
+            if (!$input->getOption('force')) {
+                $output->writeln('WARNING! This command will drop the existing index and rebuild it from scratch. To proceed, run with "--force" option.');
+
+                return;
+            }
+
+            $output->writeln(sprintf('Dropping and creating "%s" ElasticSearch index', $this->manager->getIndexName()));
+            $this->manager->dropAndCreateIndex();
 
-        $this->manager->dropAndCreateIndex();
+            $productDocumentsCreated = 0;
 
-        $productDocumentsCreated = 0;
+            /** @var ProductInterface[] $products */
+            $products = $this->productRepository->findAll();
 
-        /** @var ProductInterface[] $products */
-        $products = $this->productRepository->findAll();
-        foreach ($products as $product) {
-            $channels = $product->getChannels();
+            $output->writeln(sprintf('Loading %d products into ElasticSearch', count($products)));
 
-            /** @var ChannelInterface $channel */
-            foreach ($channels as $channel) {
-                $locales = $channel->getLocales();
-                foreach ($locales as $locale) {
-                    $productDocument = $this->productDocumentFactory->createFromSyliusSimpleProductModel(
-                        $product,
-                        $locale,
-                        $channel
-                    );
+            foreach ($products as $product) {
+                $channels = $product->getChannels();
 
-                    $this->manager->persist($productDocument);
+                /** @var ChannelInterface $channel */
+                foreach ($channels as $channel) {
+                    $locales = $channel->getLocales();
+                    foreach ($locales as $locale) {
+                        $productDocument = $this->productDocumentFactory->create(
+                            $product,
+                            $locale,
+                            $channel
+                        );
 
-                    ++$productDocumentsCreated;
-                    if (($productDocumentsCreated % 100) === 0) {
-                        $this->manager->commit();
+                        $this->manager->persist($productDocument);
+
+                        ++$productDocumentsCreated;
+                        if (($productDocumentsCreated % 100) === 0) {
+                            $this->manager->commit();
+                        }
                     }
                 }
             }
-        }
 
-        $this->manager->commit();
+            $this->manager->commit();
 
-        $output->writeln('Product index was reset!');
+            $output->writeln('Product index was rebuilt!');
+        } else {
+            $output->writeln(sprintf('<info>Command is already running</info>'));
+        }
+        $lockHandler->release();
```

Comment:
Shouldn't it release the lock only if locking itself was successful?

Reply:
Technically yes, but I think I took this example straight from the Symfony docs and it never gave me any issues or errors.